examples/dgcnn_segmentation.py

```diff
@@ -80,24 +80,16 @@ def train():
 def test(loader):
     model.eval()
 
-    y_mask = loader.dataset.y_mask
-    ious = [[] for _ in range(len(loader.dataset.categories))]
-
+    iou = 0
     for data in loader:
         data = data.to(device)
-        pred = model(data).argmax(dim=1)
-
-        i, u = i_and_u(pred, data.y, loader.dataset.num_classes, data.batch)
-        iou = i.cpu().to(torch.float) / u.cpu().to(torch.float)
-        iou[torch.isnan(iou)] = 1
+        out = model(data).argmax(dim=1)
 
-        # Find and filter the relevant classes for each category.
-        for iou, category in zip(iou.unbind(), data.category.unbind()):
-            ious[category.item()].append(iou[y_mask[category]])
+        sizes = (data.ptr[1:] - data.ptr[:-1]).tolist()
+        for pred, y in zip(out.split(sizes), data.y.split(sizes)):
+            iou += float(jaccard_index(pred, y))
```

Comment:
@Borda in v0.8 the num_classes will be required in `jaccard_index`
```suggestion
            iou += float(jaccard_index(pred, y, num_classes=loader.dataset.num_classes))
```